PR: Use rever for release process

goanpeca opened the pull request:
Fixes #51

saulshanabrook commented on rever.xsh:
```diff
@@ -0,0 +1,270 @@
+    Install release and test dependencies.
+    """
+    try:
+        conda env remove --name $TEMP_ENV -y
```
Woah that's cool you can do this in xonsh!

goanpeca commented:
Yes ! Let me make that fix

saulshanabrook commented:
Ah one other thing, would it be possible to just tell rever if you want major, minor, or patch and have it figure out the next version and bump to that?

goanpeca commented:
You mean for the current or for the next version?

saulshanabrook commented:
I meant for the next version. You run `rever major` and it releases a new major version. Now I am not sure what you would then set the version string to be after the release. 

This is something I have always been confused about with automatically switching to the .dev prefix.

If you just release version 4.0.0, then do you set the version string to?

1. 5.0.0.dev
2. 4.1.0.dev
3. 4.0.1.dev

It seems like it requires you to know, a-priori, what the next bump will be. 🤷‍♂

goanpeca commented:
I see. It is algo weird to leave the versions as the latest one you released in the package while it is in master. I will meditate on this

saulshanabrook commented:
Yeah I notice people don't like just keeping the old version.

goanpeca commented:
We can simply agree that after a version is released we add the dev without buno. Then we can do what you suggest ;-). Problem solved :-p

saulshanabrook commented:
Fine with me! :)

goanpeca commented:
>One thing I like to do is manually run some tests before actually uploading the versions, by opening JupyterLab and running some files. Is this possible with the rever setup? If so, could you document how to do that in the new RELEASE.md?

Fixed!

>Ah one other thing, would it be possible to just tell rever if you want major, minor, or patch and have it figure out the next version and bump to that?

Fixed!

I tested the steps individually but did not do a full run. So this PR will be tested when we make an official release. There might be some nitpicks.

Also could you add me as collaborator or something here

https://www.npmjs.com/package/ibis-vega-transform
https://www.npmjs.com/~goanpeca

Thanks!

saulshanabrook commented:
Thank you!

> Also could you add me as collaborator or something here

Done!


I tried running `rever major` but it seems like maybe the version string isn't being computed properly?

```
Enter your username: Uploading distributions to https://test.pypi.org/legacy/
saulshanabrook
Enter your password:
Uploading ibis_vega_transform-_4.0.0_-py3-none-any.whl
 23%|███████████████████████▉                                                                              | 8.00k/34.1k [00:00<00:02, 9.00kB/s]HTTPError: 400 Client Error: '-4.0.0-' is an invalid value for Version. Error: Start and end with a letter or numeral containing only ASCII numeric and '.', '_' and '-'. See https://packaging.python.org/specifications/core-metadata for url: https://test.pypi.org/legacy/
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 34.1k/34.1k [00:01<00:00, 27.6kB/s]
NOTE: Try --verbose to see response content.
```

goanpeca commented:
Ok, I know the problem, some extra quotes :-p 

Will fix!

goanpeca commented:
@saulshanabrook could you try again? (I had a small bug in the setup.py)
